packages/web/src/pages/trending-page/components/desktop/TrendingGenreFilters.tsx

```diff
@@ -36,48 +34,55 @@ const TrendingGenreFilters = ({
   >(null)
 
   // Avoid mutating initialGenres
-  const content = [...initialGenres]
+  const genres = [...initialGenres]
 
   // Set the last seen modal selected genre, and make sure we render it
   // even if it's not selected.
-  if (isSelectedFromModal(genre) && genre !== lastModalSelectedGenre) {
-    setLastModalSelectedGenre(genre)
-    content.push(genre)
+  if (
+    isSelectedFromModal(currentGenre) &&
+    currentGenre !== lastModalSelectedGenre
+  ) {
+    setLastModalSelectedGenre(currentGenre)
+    genres.push(currentGenre)
   } else if (lastModalSelectedGenre !== null) {
-    content.push(lastModalSelectedGenre)
+    genres.push(lastModalSelectedGenre)
   }
 
-  const didClickPill = (index: number) => {
-    // Check if we hit all
-    if (index === 0) {
-      didSelectGenre(null)
-      return
-    }
-
-    // Check if we hit overflow
-    if (index === content.length) {
-      didSelectMore()
-      return
-    }
-
-    // Select a new genre
-    didSelectGenre(content[index])
-  }
-
-  const selectedIndex = genre === null ? 0 : content.indexOf(genre)
+  const handleChange = useCallback(
+    (e: ChangeEvent<HTMLInputElement>) => {
+      const selectedGenre = e.target.value
+      if (selectedGenre === messages.all) {
+        didSelectGenre(null)
+      } else {
+        didSelectGenre(selectedGenre)
+      }
+    },
+    [didSelectGenre]
+  )
 
   return (
-    <SelectablePills
-      content={[
-        ...content.map(getCanonicalName),
-        <div key='horizontal' className={styles.overflow}>
-          <IconKebabHorizontal />
-        </div>
-      ]}
-      onClickIndex={didClickPill}
-      selectedIndex={selectedIndex}
-    />
+    <Flex gap='s' role='radiogroup' onChange={handleChange}>
+      {genres.map((genre) => (
+        <SelectablePill
+          key={genre}
+          type='radio'
+          label={getCanonicalName(genre)}
+          value={genre}
+          size='large'
+          isSelected={
+            currentGenre === null
+              ? genre === messages.all
+              : genre === currentGenre
+          }
+        />
+      ))}
```

Comment:
nit: Just an idea here (mostly my preference anyways), but I think the code would read a little more logically if we removed the "All" category from the array and statically rendered the "All" category the same way you have the "More" one rendered below
I.e.
```tsx
<SelectablePill /> // all
{genres.map(genre=>...etc)} // individual genres
<SelectablePill /> // more
```
Could then separate the `onChange` handler for the "all" pill and then the individual pill change handler would be a little simpler.

Either way, your version is already an improvement over the last one 🙌

Reply:
nice thinking, will dig into that option